### ISSUE TITLE
False positive in `prefer-top-level-await`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2177]
Should not report on following cases.

```js
async function getStat() {}

const [core, pure, bundle] = await Promise.all([
  getStat('core-js'),
  ALL && getStat('core-js-pure'),
  ALL && getStat('core-js-bundle'),
]);
```

https://github.com/zloirock/core-js/blob/ab5f0252fcbfb58aebd636d7c3b6585b2f4cc320/scripts/downloads-by-versions.mjs#L18-L24
